coderd/database/migrations/000250_built_in_psk_provisioner_keys.up.sql

```diff
@@ -0,0 +1,5 @@
+INSERT INTO provisioner_keys (id, created_at, organization_id, name, hashed_secret, tags) VALUES ('11111111-1111-1111-1111-111111111111'::uuid, NOW(), (SELECT id FROM organizations WHERE is_default = true), 'built-in', ''::bytea, '{}');
+INSERT INTO provisioner_keys (id, created_at, organization_id, name, hashed_secret, tags) VALUES ('22222222-2222-2222-2222-222222222222'::uuid, NOW(), (SELECT id FROM organizations WHERE is_default = true), 'user-auth', ''::bytea, '{}');
+INSERT INTO provisioner_keys (id, created_at, organization_id, name, hashed_secret, tags) VALUES ('33333333-3333-3333-3333-333333333333'::uuid, NOW(), (SELECT id FROM organizations WHERE is_default = true), 'psk', ''::bytea, '{}');
+
+ALTER TABLE provisioner_daemons ADD COLUMN key_id UUID REFERENCES provisioner_keys(id) ON DELETE CASCADE DEFAULT '11111111-1111-1111-1111-111111111111'::uuid NOT NULL;
```

Comment:
We have external provisioners running today on some customer deployments. Will this be ok if they are assigned to the built in?

Should we assign the existing rows to the `psk` at `33333333-3333-3333-3333-33333333333`, and have the startup code fix the built in rows?

Reply:
So I did try to consider the outputs here and picked this one since it felt like the least impact, but let me know if you feel otherwise:
- There's no data in the DB that lets us know if it's a built-in or PSK today
- External provisioners will need to reconnect after a coderd upgrade anyways, and on reconnection they will be assigned the correct `333...` id. 
- On the average deployment we would most likely see more built-ins than external, so if we have to pick one, built-in is probably less incorrect
- I considered allowing null and not back porting the reference but that felt extra messy in the resulting queries and handler logic.

So basically with this migration the external provisioners will be incorrectly labeled as built-in until the provisioners reconnect. Because we only show recent provisioners the ones that do not connect again (and stay incorrectly labeled) will only be shown for a short while before being removed from the list we show users.

Reply:
Also worth noting that right now this key reference is only for grouping in read operations, no mutation logic is performed based on this value today.